packages/playground/data-liberation/src/export/WP_Entity_Export_Iterator.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+// @TODO: This is just a hack for easy manual testing. Remove it before merge.
+add_action('plugins_loaded', function() {
+	if (!str_starts_with($_SERVER['REQUEST_URI'], '/_test_export_entities')) {
+		return;
+	}
+
+	header('Content-Type: text/plain');
+	$export_iterator = new WP_Entity_Export_Iterator();
+	foreach ($export_iterator as $key => $entity) {
+		echo "Key: $key\n";
+		var_dump($entity);
+	}
+	die();
+});
+
+// @TODO: Move to dedicated file
+class WP_Export_Entity {
+	// @TODO: If we want to naively stream entities, perhaps sites should be able
+	//        run a process that cleans up cruft like orphaned term relationships.
+	//        Then that process could be run before attempting export.
+	// @TODO: Is there any reason we need to model specific concepts like
+	//       "category" rather than just exporting terms and taxonomy
+	//       relationships?
+	//       Intuition: Directly representing terms, taxonomies,
+	//       and relationships will be more flexible at this level.
```

Comment:
Any way we slice it, let's make sure there's 100% overlap between the entities we know how to export and the entities we know how to import.